### PR TITLE
Adding Node exporter

### PR DESCRIPTION
**Pull request by rafsalas19**
- Created parent class base exporter that will be used as a parent class for any custom metrics
- Created node exporter which serves as an example of how to export custom metrics. This will export front end network tx/rx
- Modified ansible and Prometheus files to integrate the node exporter to Moneo. (Also serves as an example)
- Added unit test for node exporter
- Move Moneo docs to their own directory (docs)
- Created/ modified documentation for custom exporter creation

### CODE REVIEW

File: src/worker/shutdown.sh [hunk 1]
```diff
@@ -5,12 +5,14 @@ arch=$1
...
 	pkill -f "nvidia_exporter.py*" && sleep 1
 	pkill -f "net_exporter.py*" && sleep 1
+	pkill -f "node_exporter.py*" && sleep 1
 	pkill -f "^nv-hostengine"
 	sleep 3
 	exit 0
 elif [ $arch == "amd" ]; then
 	pkill -f "amd_exporter.py*" && sleep 1
 	pkill -f "net_exporter.py*" && sleep 1
+	pkill -f "node_exporter.py*" && sleep 1
```
Comment (yakovdyadkin) [inline src/worker/shutdown.sh:15]: Is it possible to group the common exporters to reduce duplicate lines? Maybe something like this:

```
kill_exporters() {
        pkill -f "${1}_exporter.py*" && sleep 1
	pkill -f "net_exporter.py*" && sleep 1
	pkill -f "node_exporter.py*" && sleep 1
}

if [ $arch == "nvidia" ]; then
	kill_exporters $arch
	pkill -f "^nv-hostengine"
	sleep 3
	exit 0
elif [ $arch == "amd" ]; then
        kill_exporters $arch
	pkill -f "^/opt/rocm/rdc/bin/rdcd"
	sleep 3
        exit 0
else
	echo "Proper architecture not provided. Choices are nvidia and amd."
fi
	exit 0
```

Comment (rafsalas19) [reply src/worker/shutdown.sh]: addressed

File: docs/CustomMetrics.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+How to export my own custom metrics
+=====
+Description
+-----
+Moneo provides a way for you to export your own custom node metrics. This will require you to modify/create your own custom exporter script. The [BaseExporter](./src/worker/exporters/base_exporter.py) parent class is provided for so that a child class can be created to export your custom metrics.
+
+Additionally a [NodeExporter](./src/worker/exporters/node_exporter.py) class is provided as an example of how to create a custom exporter. You may choose to modify thisas it is already integrated to the rest of the Moneo deployment process. If you choose to to create a new exporter you will have some additional modifications you will need to make (explained below).
```
Comment (jithinjosepkl) [inline docs/CustomMetrics.md:7]: typo - thisas

File: docs/CustomMetrics.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Steps
+-----
+1. If you choose to use/modify the node exporter example provided skip to step 6, otherwise continue to step 2.
+2. Create a child class for your custom metrics. Ensure to include the BaseExporter as you parent class.
+3. You will need to create 3 functions within the class. Class functions needed:
+   -  \_\_init\_\_(self, node_fields, exp_config)
+       - This function will initialize the class and the parent class. You must pass in a list of your metrics and a config dictionary (see NodeExporter for example).
+   -   collect(self, field_name)
+       - This function will be what is actuallycollecting your metrics. For example it may open a file and parse it for a give metric.
```
Comment (jithinjosepkl) [inline docs/CustomMetrics.md:17]: typo: actuallycollecting

File: docs/CustomMetrics.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+How to export my own custom metrics
+=====
+Description
+-----
+Moneo provides a way for you to export your own custom node metrics. This will require you to modify/create your own custom exporter script. The [BaseExporter](./src/worker/exporters/base_exporter.py) parent class is provided for so that a child class can be created to export your custom metrics.
```
Comment (jithinjosepkl) [inline docs/CustomMetrics.md:5]: Should we add an example use case?

File: src/worker/exporters/node_exporter.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
+######################################################
+# This exporter file can be modified or used as
+# an example of how to create a custom exporter.
```
Comment (jithinjosepkl) [inline src/worker/exporters/node_exporter.py:3]: Should we also add a high level overview of what this example exporter does? And, then highlight that a user can just edit the relevant data collection/parsing part to get custom metrics?

File: src/worker/exporters/node_exporter.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+# You can just copy paste this function.
+def init_signal_handler():
+    '''Handles exit signals, User defined signale defined in Base class'''
+    def exit_handler(signalnum, frame):
+        config['exit'] = True
+    signal.signal(signal.SIGINT, exit_handler)
+    signal.signal(signal.SIGTERM, exit_handler)
+
+
+# You can just copy paste this function
```
Comment (jithinjosepkl) [inline src/worker/exporters/node_exporter.py:116]: what might be a scenario where user will need to use this function?

**Review by jithinjosepkl — APPROVED** [review]

File: docs/CustomMetrics.md (hunk 4 above)
Comment (rafsalas19) [reply docs/CustomMetrics.md:5]: I meant for the front-end network data collection within the node_exporter to be the example case. In the instructions it says what to look at within the node_exporter

File: docs/CustomMetrics.md (hunk 3 above)
Comment (rafsalas19) [reply docs/CustomMetrics.md]: addressed

File: src/worker/exporters/node_exporter.py (hunk 5 above)
Comment (rafsalas19) [reply src/worker/exporters/node_exporter.py:3]: Like with inline comments?

File: src/worker/exporters/node_exporter.py (hunk 6 above)
Comment (rafsalas19) [reply src/worker/exporters/node_exporter.py:116]: The get _log_level function is mainly used for the arg parser to help parse log level.

File: docs/CustomMetrics.md (hunk 2 above)
Comment (rafsalas19) [reply docs/CustomMetrics.md]: addressed

File: src/worker/exporters/node_exporter.py (hunk 5 above)
Comment (jithinjosepkl) [reply src/worker/exporters/node_exporter.py:3]: I was thinking like an overview comment as well as some inline comments in specific section.

File: docs/CustomMetrics.md (hunk 4 above)
Comment (rafsalas19) [reply docs/CustomMetrics.md:5]: I will mark this resolved and work on addressing this in another comment

File: src/worker/exporters/node_exporter.py (hunk 5 above)
Comment (rafsalas19) [reply src/worker/exporters/node_exporter.py:3]: I'll address this

Comment (rafsalas19) [reply src/worker/exporters/node_exporter.py:3]: Alright I have added additional inline comments to the exporter code and an overview of the Node exporter to the documentation.

**Pull request merged**